piezo_web_app/PiezoWebApp/tests/services/spark_job/validation/argument_validator_test.py

```diff
@@ -4,6 +4,26 @@
 from PiezoWebApp.src.services.spark_job.validation import argument_validator
 
 
+@pytest.mark.parametrize("label", ["test", "label@@@@A", "12ewq", "some_longer_label"])
+def test_validate_label_validates_non_empty_strings_which_are_not_all(label):
+    # Arrange
+    validation_rule = ValidationRule({'classification': 'Optional'})
+    # Act
+    validation_result = argument_validator.validate("label", label, validation_rule)
+    # Assert
+    assert validation_result.is_valid is True
+
+
+@pytest.mark.parametrize("label", [" ", "", "     ", "all", "ALL", "All"])
+def test_validate_label_rejects_empty_strings_or_all(label):
```

Comment:
Please separate out these two scenarios (empty vs all) and check that the validation result message is appropriate.

Reply:
Method name needs updating now that "all" is removed from the parameters ;-)